WordPress/src/main/java/org/wordpress/android/util/SmartToast.java

```diff
@@ -10,51 +10,123 @@
+        AppPrefs.setInt(keyNumTimesUsed, numTypesFeatureUsed);
+        if (numTypesFeatureUsed <= MIN_TIMES_TO_USE_FEATURE) {
+            return false;
+        }
+
+        // if we're showing the toast explaining long press multiselect in the WP media library, disable showing the
+        // multiselect toast in the photo picker (and vice versa) since explaining one should explain the other
+        if (type == SmartToastType.WP_MEDIA_LONG_PRESS) {
+            disableSmartToast(SmartToastType.PHOTO_PICKER_LONG_PRESS);
+        } else if (type == SmartToastType.PHOTO_PICKER_LONG_PRESS) {
+            disableSmartToast(SmartToastType.WP_MEDIA_LONG_PRESS);
+        }
+
         int stringResId;
         switch (type) {
+            case WP_MEDIA_LONG_PRESS:
             case PHOTO_PICKER_LONG_PRESS:
-                keyCounter = UndeletablePrefKey.SMART_TOAST_PHOTO_PICKER_LONG_PRESS_COUNTER;
-                stringResId = R.string.smart_toast_photo_long_press;
-                break;
-            case WP_MEDIA_BROWSER_LONG_PRESS:
-                keyCounter = UndeletablePrefKey.SMART_TOAST_WP_MEDIA_BROWSER_LONG_PRESS_COUNTER;
                 stringResId = R.string.smart_toast_photo_long_press;
                 break;
             case COMMENTS_LONG_PRESS:
-                keyCounter = UndeletablePrefKey.SMART_TOAST_COMMENTS_LONG_PRESS_COUNTER;
                 stringResId = R.string.smart_toast_comments_long_press;
                 break;
             default:
                 return false;
         }
-        int numTimesShown = AppPrefs.getInt(keyCounter);
-        if (numTimesShown >= MAX_TIMES_TO_SHOW) {
-            return false;
-        }
 
         int yOffset = context.getResources().getDimensionPixelOffset(R.dimen.smart_toast_offset_y);
         Toast toast = Toast.makeText(context, context.getString(stringResId), Toast.LENGTH_LONG);
         toast.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.BOTTOM, 0, yOffset);
         toast.show();
 
         numTimesShown++;
-        AppPrefs.setInt(keyCounter, numTimesShown);
+        AppPrefs.setInt(keyNumTimesShown, numTimesShown);
         return true;
     }
+
+    /*
+     * prevent the passed smart toast type from being shown by setting its counter to the max - this should be
+     * used to disable long press toasts when the user long presses to multiselect since they alraady know
+     * they can do it
+     */
+    public static void disableSmartToast(@NonNull SmartToastType type) {
+        try {
+            UndeletablePrefKey key = getNumTimesToastShownKey(type);
+            AppPrefs.setInt(key, MAX_TIMES_TO_SHOW_TOAST);
+        } catch (IllegalArgumentException e) {
+            AppLog.e(AppLog.T.MEDIA, e);
+        }
+    }
+
+    /*
+     * get the preference key which stores the number of times the feature associated with the smart toast type has
+     * been used
+     */
+    private static UndeletablePrefKey getNumTimesFeatureUsedKey(@NonNull SmartToastType type) {
```

Comment:
How about introducing a field in `SmartToastType` enum itself, holding the key? That should remove repetition and remove the need to check for the `default` case. [Example implementation.](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/HelpshiftHelper.java#L42-L55)

Same for `getNumTimesToastShownKey()`.